lynx/src/main/java/com/github/pedrovgs/lynx/model/Lynx.java

```diff
@@ -133,9 +133,18 @@ private boolean shouldAddTrace(String logcatTrace) {
   }
 
   private boolean traceMatchesFilter(String logcatTrace) {
+    TraceLevel levelFilter = lynxConfig.getFilterTraceLevel();
     String filter = lynxConfig.getFilter().toLowerCase();
     String logcatTraceLowercase = logcatTrace.toLowerCase();
-    return logcatTraceLowercase.contains(filter);
+    return logcatTraceLowercase.contains(filter) && containsTraceLevel(logcatTraceLowercase,
+        levelFilter);
+  }
+
+  private boolean containsTraceLevel(String logcatTraceLowercase, TraceLevel levelFilter) {
+    return levelFilter.equals(TraceLevel.ALL)
+        || logcatTraceLowercase.charAt(Trace.TRACE_LEVEL_INDEX) == levelFilter.getValue()
+        .toLowerCase()
+        .charAt(0);
```

Comment:
Cool! This implementation is faster than wait to have a Trace object and then check if pass the filter or not! Remember to change it with the new implementation where, for example, an error trace should pass the filter if a warning trace level has been configured as filter.

![The Flash](http://media2.giphy.com/media/F3KhL2UHkBCnu/200.gif)